src/main/java/com/blackduck/integration/detect/workflow/project/ProjectNameVersionDecider.java

```diff
@@ -37,7 +37,7 @@ public NameVersion decideProjectNameVersion(List<DetectTool> preferredDetectTool
         String decidedProjectVersionName;
         if (StringUtils.isNotBlank(projectVersionOptions.overrideProjectVersionName)) {
             decidedProjectVersionName = projectVersionOptions.overrideProjectVersionName;
-        } else if (chosenToolVersion.isPresent() && !chosenToolVersion.get().isBlank()) {
+        } else if (chosenToolVersion != null && chosenToolVersion.isPresent() && !chosenToolVersion.get().trim().isEmpty()) {
```

Comment:
nit: At line 38 we're already using `StringUtils.isNotBlank()` for similar purposes. It might be good to do the same here for consistency.

Reply:
Yeah I think this is a good approach to cleanup the harder to read long check. Moving to this.